CONTRIBUTING.md

```diff
@@ -1,102 +1,181 @@
+```
+npm run start
+```
 
-- Can I access something that's not mine, or something I shouldn't have access to?
-- Can I disable something for other people?
+## Working on fixes or features
 
-If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at security@sanity.io.
+Please create a new branch branch for the fix or feature/enhancement you want to work on and regularly push your work to this branch. Remember to also keep your branch up to date by rebasing it regularly against the original `next` branch. To do this, add the original Sanity repository as a remote (in our example we called it `upstream`).
 
-## How to report a bug
+```
+git remote add upstream git@github.com:sanity-io/sanity.git
+git pull upstream next --rebase
+```
 
-When filing an issue, make sure to answer these six questions:
+### New features
 
-- Which versions of Sanity are you using (`sanity versions`)?
-- What operating system are you using?
-- Which versions of Node.js / npm are you running?
-- What did you do?
-- What did you expect to see?
-- What did you see instead?
+Please note that if you would like to develop a new feature for the Sanity Studio, it's unlikely that we will be able to accept your contribution. However, we highly recommend that you develop and publish your new feature as a plugin. Plugins can be [parts](https://www.sanity.io/docs/extending/parts), [custom input components](https://www.sanity.io/docs/extending/custom-input-widgets), [dashboard widgets](https://www.sanity.io/docs/dashboard/creating-your-own-widget), and [Studio themes](https://www.sanity.io/docs/guides/how-to-brand-your-studio) that you and others can use to customize and extend the functionality of the Studio.
 
-## How to suggest a feature or enhancement
+# General info
 
-If you find yourself wishing for a feature that doesn't exist in Sanity, you are probably not alone. There are bound to be others out there with similar needs. Many of the features that Sanity has today have been added because our users saw the need. Open an issue on our issues list on GitHub which describes the feature you would like to see, why you need it, and how it should work.
+- Anything in the `next` branch is scheduled for the next release
+- To work on a fix or something new, create a descriptively named branch off of `next` (ie: feature/new-oauth2-scopes)
+- Commit to that branch locally and regularly push your work to the same named branch
+- Rebase your feature branch regularly against `next`. Make sure its even with `next` before creating a Pull Request or merging
+- Pull Requests should be as ready as possible for merge and the Pull Request template should be filled out to the best of your ability or in a way that is appropriate. Unless stated otherwise, it should also be safe to assume that:
```

Comment:
I this this is somewhat wrong. It's possible to submit a draft PR. This to receive feedback on the ongoing development of a fix or similar. I think we should encourage people to ask us for guidance when they need it. There might be a lot of questions regarding the code base when working on a fix. Like Espen said, it's not good that they spend a lot of time and submitting a PR at the end, and if it isn't solved the way we want, we need to decline it. I think it's highly unlikely that they would submit another one after that.